sdk/core/core-util/src/datetime.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+export type OffsetMillisecondsPrecision = 0 | 1 | 2 | 3;
+
+export function toOffsetDateTime(date: Date, precision: OffsetMillisecondsPrecision = 0): string {
+  const tzOffset = -date.getTimezoneOffset();
+  const diff = tzOffset >= 0 ? "+" : "-";
+  const pad = (n: number) => `${Math.floor(Math.abs(n))}`.padStart(2, "0");
+  const precisedMilliseconds = getPrecisedMilliseconds(date, precision);
+  return (
+    date.getFullYear() +
```

Comment:
I was wondering if this implementation would work if timezone is crossed year for example utc is in 2023, but with offset timezone it may 2022. The same question for crossed month and day.